### DISCUSSION TITLE
Flash loans

### DISCUSSION THREAD

**Discussion by PaulRBerg** [sablier-labs/flow#23] [Ideas]
We have recently [removed flash loans](https://github.com/sablier-labs/v2-core/pull/779) from the Lockup series because they can be a [deterrent](https://github.com/sablier-labs/v2-core/discussions/278) for vesting users.

However, none of those arguments apply to OpenEnded, which is meant to be used with stablecoins.

Any thoughts on enabling support for flash loans in OpenEnded, @sablier-labs/solidity + @razgraf?

**Comment by andreivladbrg** [comment]
Interesting proposal.

I don't have a particular opinion at the moment.

Some points that come to mind now:
- deposits are not a requirement, meaning the moments when the contracts actually hold the assets will probably be less frequent (resulting in fewer profits)
- one Sablier smart contract project offers flash loans, and another does not. This could lead to confusion among users and might "scare" them away (the vesting ones)
- technically, it shouldn't be a problem
- this should be more of a business decision

**Reply by PaulRBerg** [reply]
Thanks for your feedback.

Deposits not being required is just a feature - users will still be compelled to deposit stablecoins. Compared to Lockup, the average deposit per user will be smaller, but there will be more deposits (and more users, hopefully), which makes me optimistic that the stablecoin TVL will be greater in OpenEnded.

For instance, LlamaPay has [millions in TVL](https://defillama.com/protocol/llamapay) at any one time.
  
> This could lead to confusion among users and might "scare" them away (the vesting ones)

I think you're over-estimating how many users look at the underlying smart contract design.

**Reply by andreivladbrg** [reply]
>LlamaPay has [millions in TVL](https://defillama.com/protocol/llamapay) at any one time

most of the TVL is from the vesting contract not the payroll one

> think you're over-estimating how many users look at the underlying smart contract design

probably

**Reply by PaulRBerg** [reply]
Yes but even without vesting, they still have $3M.

I agree that that is not a lot, though.

**Comment by razgraf** [comment]
I would leave flash loans out of the protocol for now on account of: simplicity and speed to go-to-market.

We can re-evaluate them at a later date, based on the median stablecoin TVL and fees market for flash loan providers.

**Reply by PaulRBerg** [reply]
Fair.

The development cost is small because we've already built and audited flash loans.

I think the stronger argument against flash loans is opportunity cost.

Lending user assets + SabVM matter infinitely more.

Closing.
